From 029037c4693bff6a9e0062cbfdaf4055826604d1 Mon Sep 17 00:00:00 2001
From: ghost <andreas.monzner@dream-property.net>
Date: Thu, 17 Jan 2013 14:37:02 +0100
Subject: [PATCH] tools/enigma2.sh.in: set the linux OOM score for the enigma2
 process.. so the kernel first kills other processes on low
 memory conditions...

---
 tools/enigma2.sh.in |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/tools/enigma2.sh.in b/tools/enigma2.sh.in
index 5644d14..fc3a979 100755
--- /usr/bin/enigma2.sh
+++ /usr/bin/enigma2.sh
@@ -30,6 +30,9 @@ if [ -e /usr/lib/libpassthrough.so ]; then
 	LIBS="$LIBS /usr/lib/libpassthrough.so"
 fi
 
+(sleep 2; echo "enigma2 is the main pvr application... adjust oom score!"; PID=$(pidof enigma2); \
+	[ -e /proc/$PID/oom_score_adj ] && echo "-999" > /proc/$PID/oom_score_adj || echo "-17" > /proc/$PID/oom_adj;) &
+
 PAGECACHE_FLUSH_INTERVAL=$((512*1024)) LD_PRELOAD=$LIBS /usr/bin/enigma2
 
 # enigma2 exit codes:
-- 
1.7.10.4

